### ISSUE TITLE
AssertionError: undefined == true

### ISSUE THREAD

**Issue by jmls** [strongloop/loopback#1324]
we are getting this error since the commit https://github.com/strongloop/loopback/commit/b61fae58f6b810ac18df5f3baa2d0aec73305ed4

in our model.js code, we have the following

```javascript
 module.exports = function(MyModel) {
    MyModel.getApp(function(err, app) {
```

this now generates the error 

assert.js:86
  throw new assert.AssertionError({
        ^
AssertionError: undefined == true
    at /opt/nodegloo/node_modules/loopback/lib/model.js:392:7
    at EventEmitter.<anonymous> (/opt/nodegloo/node_modules/loopback/lib/model.js:222:9)
    at EventEmitter.g (events.js:199:16)
    at EventEmitter.emit (events.js:107:17)
    at EventEmitter.app.model (/opt/nodegloo/node_modules/loopback/lib/application.js:157:9)
    at /opt/nodegloo/node_modules/loopback-boot/lib/executor.js:171:9
    at Array.forEach (native)
    at setupModels (/opt/nodegloo/node_modules/loopback-boot/lib/executor.js:167:23)
    at execute (/opt/nodegloo/node_modules/loopback-boot/lib/executor.js:32:3)
    at bootLoopBackApp (/opt/nodegloo/node_modules/loopback-boot/index.js:129:3)

**Comment by bajtos** [comment]
Can you please provide us with a simple application reproducing the problem?

Try to change `getApp` to the following, does it fix the problem for you?

```js
  Model.getApp = function(callback) {
    var self = this;
    self._runWhenAttachedToApp(function(app) {
      assert(self.app);
      callback(null, self.app);
    });
  };
```

**Comment by jmls** [comment]
jeez ;) Simple example

 slc loopback (accept all default)
 slc loopback:model (called Bar, one property called field1, type string)
 node .

this works

```javascript 
vi common/models/bar.js
```
add

```javascript 
Bar.getApp(function(err, app) {
 })
```
start app again

node . 

now crashes with error 

```javascript
assert.js:86
  throw new assert.AssertionError({
        ^
AssertionError: undefined == true
    at /tmp/foo/node_modules/loopback/lib/model.js:392:7
    at EventEmitter.<anonymous> (/tmp/foo/node_modules/loopback/lib/model.js:222:9)
    at EventEmitter.g (events.js:199:16)
    at EventEmitter.emit (events.js:107:17)
    at EventEmitter.app.model (/tmp/foo/node_modules/loopback/lib/application.js:157:9)
    at /tmp/foo/node_modules/loopback-boot/lib/executor.js:169:9
    at Array.forEach (native)
    at setupModels (/tmp/foo/node_modules/loopback-boot/lib/executor.js:165:23)
    at execute (/tmp/foo/node_modules/loopback-boot/lib/executor.js:32:3)
    at bootLoopBackApp (/tmp/foo/node_modules/loopback-boot/index.js:129:3)
```

modifying the code as suggested makes the error go away

trust that this is enough info for you ;)

**Comment by bajtos** [comment]
Thanks. Here is a minimal app that reproduced the problem on my machine:

```js
var loopback = require('./');
var app = loopback();

var Bar = loopback.createModel('Bar');
Bar.getApp(function(err, app) { console.log('PASS'); });

app.dataSource('db', { connector: 'memory' });
app.model(Bar, { dataSource: 'db' });
```

**Comment by jmls** [comment]
if it's reproducible, it's fixable, right ? ;)

**Comment by bajtos** [comment]
@jmls sure, see https://github.com/strongloop/loopback/pull/1343

**Comment by bajtos** [comment]
Fixed in `loopback@2.17.2`.
